specs/ClientCertificate.md

```diff
@@ -0,0 +1,500 @@
+                    // perform async work before the event completes.
+                    m_completeDeferredCertificateRequestedEvent = [showDialog, deferral] {
+                        showDialog();
+                        CHECK_FAILURE(deferral->Complete());
+                    };
+
+                    return S_OK;
+                })
+                .Get(),
+            &m_ClientCertificateRequestedToken));
+    //! [ClientCertificateRequested2]
+}
+```
+
+## .NET/ WinRT
+```c#
+private bool _isClientCertificateRequested = false;
+void ClientCertificateRequestedCmdExecuted(object target, ExecutedRoutedEventArgs e)
+{
+  try
+  {
+    if (!_isClientCertificateRequested)
+    {
+      webView.CoreWebView2.ClientCertificateRequested += delegate (
+      object sender, CoreWebView2ClientCertificateRequestedEventArgs args)
+      {
+        // Developer can obtain a deferral for the event so that the WebView2
+        // doesn't examine the properties we set on the event args until
+        // after the deferral completes asynchronously.
+        CoreWebView2Deferral deferral = args.GetDeferral();
+
+        System.Threading.SynchronizationContext.Current.Post((_) =>
+        {
+          using (deferral)
+          {
+            List<CoreWebView2ClientCertificate> certificatesList = args.MutuallyTrustedCertificates;
+            if (certificatesList.Count() > 0)
+            {
+              var dialog = new ClientCertificateSelectionDialog(
+                           title: "Client Certificate Selection",
+                           host: args.Host,
+                           port: args.Port,
+                           client_cert_list: certificatesList);
+              if (dialog.ShowDialog() == true)
+              {
+                args.SelectedCertificate = (CoreWebView2ClientCertificate)dialog.CertificateDataBinding.SelectedItem;
+                args.ClientCertificateRequestResponseState = CoreWebView2ClientCertificateRequestResponseState.ContinueWithCertificate;
+              }
+              else
+              {
+                args.ClientCertificateRequestResponseState = CoreWebView2ClientCertificateRequestResponseState.ContinueWithoutCertificate;
+              }
+            }
+            else
+            {
+              args.ClientCertificateRequestResponseState = CoreWebView2ClientCertificateRequestResponseState.ContinueWithoutCertificate;
+            }
+          }
+        }, null);
+      };
+      _isClientCertificateRequested = true;
+    }
+  }
+  catch (NotImplementedException exception)
+  {
+    MessageBox.Show(this, "Client Certificate Requested Failed: " + exception.Message, "Client Certificate Requested");
+  }
+}
+```
+# Remarks
+
+# API Notes
+
+# API Details
+See [API Details](#api-details) section below for API reference.
+
+## Win32 C++
+```cpp
+interface ICoreWebView2_3;
+interface ICoreWebView2ClientCertificate;
+interface ICoreWebView2CertificateAuthorityList;
```

Comment:
Should use the plural rather than a "List" or "Collection" suffix: `ICoreWebView2CertificateAuthorities`